src/groups/mqb/mqba/mqba_authenticator.h

```diff
@@ -0,0 +1,168 @@
+// Copyright 2025 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// mqba_authenticator.h                                           -*-C++-*-
+#ifndef INCLUDED_MQBA_AUTHENTICATOR
+#define INCLUDED_MQBA_AUTHENTICATOR
+
+/// @file mqba_authenticator.h
+///
+/// @brief
+///
+///
+/// Thread Safety                              {#mqba_authenticator_thread}
+/// =============
+///
+/// The implementation must be thread safe as 'authenticate()' may be called
+/// concurrently from many IO threads.
+
+// MQB
+#include <mqbconfm_messages.h>
+#include <mqbnet_authenticator.h>
+
+// BMQ
+#include <bmqio_channel.h>
+#include <bmqio_status.h>
+#include <bmqp_ctrlmsg_messages.h>
+
+// BDE
+#include <bdlbb_blob.h>
+#include <bdlcc_sharedobjectpool.h>
+#include <bsl_memory.h>
+#include <bsl_ostream.h>
+#include <bslma_allocator.h>
+#include <bslma_usesbslmaallocator.h>
+#include <bslmf_nestedtraitdeclaration.h>
+
+namespace BloombergLP {
+
+// FORWARD DECLARATION
+namespace mqbblp {
+class ClusterCatalog;
+}
+namespace mqbi {
+class Dispatcher;
+}
+
+namespace mqba {
+
+// ===================
+// class Authenticator
+// ===================
+
+/// Authenticator for a BlazingMQ session with client or broker
+class Authenticator : public mqbnet::Authenticator {
+  public:
+    // TYPES
+
+    /// Type of a pool of shared pointers to blob
+    typedef bdlcc::SharedObjectPool<
+        bdlbb::Blob,
+        bdlcc::ObjectPoolFunctors::DefaultCreator,
+        bdlcc::ObjectPoolFunctors::RemoveAll<bdlbb::Blob> >
+        BlobSpPool;
+
+  private:
+    typedef bsl::shared_ptr<mqbnet::AuthenticationContext>
+        AuthenticationContextSp;
+
+  private:
+    // DATA
+
+    /// Allocator to use.
+    bslma::Allocator* d_allocator_p;
+
+    BlobSpPool* d_blobSpPool_p;
+
+  private:
+    // NOT IMPLEMENTED
+
+    /// Copy constructor and assignment operator not implemented.
+    Authenticator(const Authenticator&);             // = delete
+    Authenticator& operator=(const Authenticator&);  // = delete
+
+  private:
+    // PRIVATE MANIPULATORS
+
+    /// Invoked when received a `ClientIdentity` authentication message with
```

Comment:
"Invoked when a `ClientIdentity` authentication message is received with the specified `context`"

Same below